### PR TITLE
updated docs with a page for operations/logging

### PR DESCRIPTION
**Pull request by mohnishbasha**
Issue:
- https://github.com/slatedb/slatedb/issues/129

Solution:
- Added page for operations/logging
- Added basic documentation on how to enable tracing within rust/ slatedb project. 

Test:
- See Screenshot
Screenshot 2024-11-13 at 3 11 49 PM

### CODE REVIEW

File: docs/operations/logging.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+## Getting Started
+
+To enable logging in your SlateDB application, you'll need to add the following dependencies to your `Cargo.toml`:
+
+```toml
+[dependencies]
+tracing = "0.1"
+tracing-subscriber = { version = "0.3", features = ["env-filter"] }
+env_logger = "0.10"
```
Comment (criccomini) [inline docs/operations/logging.md]: I don't think you need this.

File: docs/operations/logging.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+```
+
+## Basic Setup
+
+Here's a basic example showing how to initialize tracing with `env_logger` and SlateDB:
```
Comment (criccomini) [inline docs/operations/logging.md]: I don't think the example uses env_logger, does it?

File: docs/operations/logging.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+```rust
+use slatedb::Database;
+use tracing::{info, warn, error};
+use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt};
+
+#[tokio::main]
+async fn main() {
+    // Initialize tracing subscriber
+    tracing_subscriber::registry()
```
Comment (criccomini) [inline docs/operations/logging.md]: We just do [`tracing_subscriber::fmt::init();`](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/fn.init.html). I think this is roughly what you're doing.

File: docs/operations/logging.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        .with(tracing_subscriber::EnvFilter::try_from_default_env()
+            .unwrap_or_else(|_| "info".into()))
+        .with(tracing_subscriber::fmt::layer())
+        .init();
+
+    // Now SlateDB operations will emit traces
+    info!("Initializing SlateDB");
+    
+    let db = Database::new()
```
Comment (criccomini) [inline docs/operations/logging.md]: There is no `Database`. We have a `Db`. Is that what you meant?

File: docs/operations/logging.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    // Your database operations will now emit traces
+    let result = db.get("my_key").await;
+    match result {
+        Ok(value) => info!("Retrieved value successfully"),
+        Err(e) => error!("Error retrieving value: {}", e),
+    }
+}
+```
+
+## Environment Variables
```
Comment (criccomini) [inline docs/operations/logging.md]: All of this seems unrelated to SlateDB. It's just about the tracing library. Can you remove it? I think the links at the bottom should be sufficient.

**Review by criccomini — COMMENTED** [review]
Kinda seems like you used AI to generate this? The code example is referencing classes that don't even exist.

File: docs/operations/logging.md (hunk 5 above)
Comment (mohnishbasha) [reply docs/operations/logging.md]: sounds good, i took a generic tracing example and tried to apply it here. I will have it updated.

File: docs/operations/logging.md (hunk 3 above)
Comment (mohnishbasha) [reply docs/operations/logging.md]: yea my bad, the example i put in here is not very relevant. I wanted to gather some inputs on the docs layout itself so I put a example as place holder. I will remove all the extra verbiage and update the example with a more relevant code snippet (will validate it using an actual test snippet that I can run).

File: docs/operations/logging.md (hunk 4 above)
Comment (mohnishbasha) [reply docs/operations/logging.md]: +1, the example is just some placeholder code. I will have it updated with a more appropriate snippet. I was thinking I can gather some feedback on the docs layout before the final update. Should have left it in draft mode until then.

**Comment by mohnishbasha** [comment]
Question:
- Does the overall layout for the page content looks ok?

**Comment by mohnishbasha** [comment]
@criccomini please review this example using env-logger:
-  https://github.com/mohnishbasha/slatedb-system-tests/blob/master/test-tracing-with-env-logger/src/main.rs

I have updated the doc based on the above ^^. Here is how the output looks with RUST_LOG=trace:
```
Running `target/debug/test-tracing-with-env-logger`
[2024-11-14T17:08:35Z TRACE test_tracing_with_env_logger] the answer was: 12
[2024-11-14T17:08:35Z DEBUG test_tracing_with_env_logger] this is a debug message
[2024-11-14T17:08:35Z ERROR test_tracing_with_env_logger] this is printed by default
[2024-11-14T17:08:35Z INFO  test_tracing_with_env_logger] this is printed by default
[2024-11-14T17:08:35Z WARN  test_tracing_with_env_logger] this is printed by default
[2024-11-14T17:08:35Z INFO  slatedb::db] Opening database at path: Path { raw: "tmp/test_kv_store2" }
[2024-11-14T17:08:35Z INFO  slatedb::db] Using options: {
      "flush_interval": "100ms",
      "manifest_poll_interval": "1s",
      "min_filter_keys": 1000,
      "filter_bits_per_key": 10,
      "l0_sst_size_bytes": 67108864,
      "l0_max_ssts": 8,
      "max_unflushed_bytes": 1073741824,
      "compactor_options": {
        "poll_interval": "5s",
        "max_sst_size": 1073741824,
        "max_concurrent_compactions": 4
      },
      "compression_codec": null,
      "object_store_cache_options": {
        "root_folder": null,
        "max_cache_size_bytes": 17179869184,
        "part_size_bytes": 4194304,
        "scan_interval": "3600s"
      },
      "garbage_collector_options": {
        "manifest_options": {
          "poll_interval": "300s",
          "min_age": "86400s"
        },
        "wal_options": {
          "poll_interval": "60s",
          "min_age": "60s"
        },
        "compacted_options": {
          "poll_interval": "300s",
          "min_age": "86400s"
        }
      },
      "default_ttl": null,
      "max_unflushed_memtable": 1
    }

```

File: docs/operations/logging.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    debug!("this is a debug {}", "message");
+    error!("this is printed by default");
+    info!("this is printed by default");
+    warn!("this is printed by default");
+    log_enabled!(Level::Info);
+
+    // Setup
+    let object_store: Arc<dyn ObjectStore> = Arc::new(InMemory::new());
+    let options = DbOptions::default();
+    let fp_registry = Arc::new(FailPointRegistry::new());
```
Comment (criccomini) [inline docs/operations/logging.md]: No need for this. Please see the quickstart for an example.

File: docs/operations/logging.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+title: Tracing in SlateDB
+---
+
+# Tracing in SlateDB
+
+SlateDB uses the [`tracing`](https://github.com/tokio-rs/tracing) library for logging and diagnostics. This provides structured, contextual logging that helps debug and monitor your database operations.
+
+## Getting Started
+
+To enable tracing in your SlateDB application, you'll need to add the following dependencies to your `Cargo.toml`:
```
Comment (criccomini) [inline docs/operations/logging.md]: This is not quite correct. You don't need `env_logger` to enable logging. In fact, we don't include `env_logger` at all in SlateDB's dependencies and can see logs just fine using the init() command I showed earlier.

File: docs/operations/logging.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+use log::{debug, error, log_enabled, warn, info, Level, trace};
+
+
+#[tokio::main]
+async fn main() {
+
+    // enable env logger
+    env_logger::init();
+
+    // test trace log
```
Comment (criccomini) [inline docs/operations/logging.md]: These are all just showing how to use logging in general. Please remove as it's not relevant to SlateDB.

File: docs/operations/logging.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+RUST_LOG=debug cargo run
+
+# Show only info and above for SlateDB
+RUST_LOG=slatedb=info cargo run
+
+# Show debug logs for SlateDB and info for everything else
+RUST_LOG=debug,slatedb=debug cargo run
+```
+
+## Log Levels
```
Comment (criccomini) [inline docs/operations/logging.md]: Please remove as this is just generic logging that doesn't have anything to do with SlateDB.

File: docs/operations/logging.md [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+    assert!(db.get(key).await.unwrap().is_none());
+
+    // Close
+    db.close().await.unwrap();
+}
+```
+
+## Environment Variables
+
+You can control the logging level using the `RUST_LOG` environment variable:
```
Comment (criccomini) [inline docs/operations/logging.md]: Please remove as this is just generic logging that doesn't have anything to do with SlateDB.

**Review by criccomini — CHANGES_REQUESTED** [review]

File: docs/operations/logging.md (hunk 6 above)
Comment (mohnishbasha) [reply docs/operations/logging.md]: right, i should have used that instead.

File: docs/operations/logging.md (hunk 7 above)
Comment (mohnishbasha) [reply docs/operations/logging.md]: Sounds good, I will have it updated using tracing_subscriber::fmt::init(); instead of env_logger::init().

I was following the instructions from the issue comment you referenced, which mentioned using env_logger for the documentation. 
https://github.com/slatedb/slatedb/issues/129#issuecomment-2342351820

**Comment by mohnishbasha** [comment]
Updated using the quick start snippet and test it out here:
- https://github.com/mohnishbasha/slatedb-system-tests/blob/master/test-tracing-quickstart/src/main.rs

Here is the output from `cargo run` with `RUST_LOG=info`
```
Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.21s
     Running `target/debug/test-tracing-quickstart`
2024-11-14T19:06:25.808894Z  INFO slatedb::db: Opening database at path: Path { raw: "tmp/test_kv_store" }
2024-11-14T19:06:25.809078Z  INFO slatedb::db: Using options: {
  "flush_interval": "100ms",
  "manifest_poll_interval": "1s",
  "min_filter_keys": 1000,
  "filter_bits_per_key": 10,
  "l0_sst_size_bytes": 67108864,
  "l0_max_ssts": 8,
  "max_unflushed_bytes": 1073741824,
  "compactor_options": {
    "poll_interval": "5s",
    "max_sst_size": 1073741824,
    "max_concurrent_compactions": 4
  },
  "compression_codec": null,
  "object_store_cache_options": {
    "root_folder": null,
    "max_cache_size_bytes": 17179869184,
    "part_size_bytes": 4194304,
    "scan_interval": "3600s"
  },
  "garbage_collector_options": {
    "manifest_options": {
      "poll_interval": "300s",
      "min_age": "86400s"
    },
    "wal_options": {
      "poll_interval": "60s",
      "min_age": "60s"
    },
    "compacted_options": {
      "poll_interval": "300s",
      "min_age": "86400s"
    }
  },
  "default_ttl": null
}
```

**Review by criccomini — APPROVED** [review]
